### PR TITLE
Refactor to match Terraform standard module structure

### PR DESCRIPTION
**Pull request by pwideman**
This pull request includes changes to align the structure of the code to the [Terraform standard module structure](https://developer.hashicorp.com/terraform/language/modules/develop/structure).

Changes include:

- `nsg` and `firewall` modules moved to `/modules` directory, shared ARM template moved to `/modules/shared`
- Removal of Terraform lock files (`.terraform.lock.hcl`), and added to `.gitignore`
- Added examples for both the `nsg` and `firewall` modules, split into separate subdirectories in `/examples`
- Added root module, which is just a repackaging of the `nsg` module. This is temporary, the next step is to give meaning to the root module by making it an encapsulation of both submodules

### CODE REVIEW

File: examples/example-firewall/main.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Create a VNET using an Azure Firewall to control the VNET network access
+module "vnet" {
+    source = "github.com/pwideman/github-runner-vnet//modules/firewall"
```
Comment (garnertb) [inline examples/example-firewall/main.tf:3]:
```suggestion
    source = "github.com/garnertb/github-runner-vnet//modules/firewall"
```

File: examples/example-firewall/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+output "gh_network_settings_resource_id" {
+    description = "ID of the GitHub.Network/networkSettings resource"
+    value = module.vnet.network_settings_id
+}
```
Comment (garnertb) [inline examples/example-firewall/outputs.tf:4]:
```suggestion
}

```

File: examples/example-nsg/main.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Create a VNET using an Azure Network Security Group to control the VNET network access
+module "vnet" {
+    source = "github.com/pwideman/github-runner-vnet//modules/nsg"
```
Comment (garnertb) [inline examples/example-nsg/main.tf:3]:
```suggestion
    source = "github.com/garnertb/github-runner-vnet//modules/nsg"
```

File: examples/example-nsg/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+output "gh_network_settings_resource_id" {
+    description = "ID of the GitHub.Network/networkSettings resource"
+    value = module.vnet.network_settings_id
+}
```
Comment (garnertb) [inline examples/example-nsg/outputs.tf:4]:
```suggestion
}

```

File: outputs.tf [hunk 5]
```diff
@@ -0,0 +1,4 @@
+output "gh_network_settings_resource_id" {
+    description = "ID of the GitHub.Network/networkSettings resource"
+    value = module.vnet.network_settings_id
+}
```
Comment (garnertb) [inline outputs.tf:4]:
```suggestion
}

```

File: vars.tf [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  default     = ["10.0.0.0/16"]
+}
+
+variable "runner_subnet_address_prefixes" {
+  description = "Address prefixes for the runner subnet"
+  type        = list(string)
+  default     = ["10.0.0.0/24"]
+}
+
+variable "github_org_id" {
```
Comment (garnertb) [inline vars.tf:24]: Do you think we should rename this to `github_enterprise_id`?

**Review by garnertb — COMMENTED** [review]
Looks good,

**Review by garnertb — APPROVED** [review]

File: vars.tf (hunk 6 above)
Comment (pwideman) [reply vars.tf]: Agreed, done.

**Review by garnertb — APPROVED** [review]

**Pull request merged**